src/uu/pidof/src/pidof.rs

```diff
@@ -0,0 +1,161 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::path::PathBuf;
+
+use clap::{crate_version, Arg, ArgAction, ArgMatches, Command};
+use uu_pgrep::process::{walk_process, ProcessInformation};
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+const ABOUT: &str = help_about!("pidof.md");
+const USAGE: &str = help_usage!("pidof.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let arg_program_name = matches.get_many::<String>("program-name");
+    let arg_separator = matches.get_one::<String>("d").unwrap();
+
+    if arg_program_name.is_none() {
+        uucore::error::set_exit_code(1);
+        return Ok(());
+    };
+
+    let collected = collect_matched_pids(&matches);
+
+    if collected.is_empty() {
+        uucore::error::set_exit_code(1);
+        return Ok(());
+    };
+
+    let output = collected
+        .into_iter()
+        .map(|it| it.pid.to_string())
+        .collect::<Vec<_>>()
+        .join(arg_separator);
+
+    let flag_quiet = matches.get_flag("q");
+    if !flag_quiet {
+        println!("{output}");
+    }
+
+    Ok(())
+}
+
+fn get_executable_name(process: &mut ProcessInformation) -> String {
+    let binding = process.cmdline.split(' ').collect::<Vec<_>>();
+    let mut path = binding.first().unwrap().to_string();
+
+    if path.is_empty() {
+        path.clone_from(&process.status()["Name"]);
+    };
+
+    PathBuf::from(path)
+        .file_name()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_string()
+}
+
+fn collect_matched_pids(matches: &ArgMatches) -> Vec<ProcessInformation> {
+    let program_name: Vec<_> = matches
```

Comment:
Here I would use the plural `program_names` because the variable is a `Vec` and thus can contain multiple program names.